File: sample.py
import pickle
from matplotlib import pyplot as plt


data = pickle.load(open("ytc_py.pkl", 'rb'))
X_train, y_train = data['X_train'], data['y_train']
X_test, y_test = data['X_test'], data['y_test']

print("Number of training image-sets: ", len(X_train))
print("Number of testing image-sets: ", len(X_test))
print("Feature dimension of each image: ", X_train.shape)

plt.imshow(X_train[0][:, 10].reshape((20, 20)))
plt.show()

File: sample.m
clear

%% データの読み込み
load ytc_matlab.mat

%% データの情報

fprintf("学習データの画像セット数: %d\n", size(X_train, 2));
fprintf("テストデータの画像セット数: %d\n", size(X_test, 2));
fprintf("特徴量の次元: %d\n", size(X_train{1}, 1));


%% 画像の表示
colormap(gray)
imagesc(reshape(X_train{1}(:, 1), 20, 20)')

